lambdas/preprocessing/src/lambda_function.py

```diff
@@ -1,46 +1,81 @@
 
+import requests
+from requests import Response
+import json
+from raster_analysis.boto import get_secrets_manager_client
+
 patch(["boto3"])
 
+GEOSTORE_PAGE_SIZE = 25
+
+
+class UnexpectedResponseError(Exception):
+    pass
+
 
 @xray_recorder.capture("Preprocessing")
-def handler(event, context):
+def handler(event: Dict[str, Any], context: Any) -> Any:
     try:
         LOGGER.info(f"Running preprocessing with parameters: {event}")
         fc: Optional[Dict] = event.get("feature_collection")
         uri: Optional[str] = event.get("uri")
-        id_field = event.get("id_field", "fid")
+        geostore_ids: Optional[List[str]] = event.get("geostore_ids")
+        id_field: str = event.get("id_field", "fid")
 
-        if fc is not None and uri is not None:
-            raise Exception("Please specify GeoJSON via (only) one parameter!")
+        gpdf = None
+        geostore_info = None
+        if (fc and uri) or (fc and geostore_ids) or (uri and geostore_ids):
+            raise Exception("Please specify exactly one of 'feature_collection', 'uri', or 'geostore_ids'.")
         elif fc is not None:
             gpdf = gpd.GeoDataFrame.from_features(fc, columns=[id_field, "geometry"])
         elif uri is not None:
             gpdf = gpd.read_file(uri, columns=[id_field, "geometry"])
+        elif geostore_ids is not None:
+            geostore_info = get_geostore_info(geostore_ids)
         else:
-            raise Exception("Please specify GeoJSON via (only) one parameter!")
+            raise Exception("Please specify exactly one of 'feature_collection', 'uri', or 'geostore_ids'.")
 
-        if id_field not in gpdf.columns.tolist():
+        if gpdf is not None and id_field not in gpdf.columns.tolist():
             raise Exception(f"Input feature collection is missing ID field '{id_field}'")
 
-        rows = []
-        for record in gpdf.itertuples():
-            geom = shape(getattr(record, "geometry"))
-            encoded_geom = encode_geometry(geom)
-            rows.append([getattr(record, id_field), encoded_geom])
+        rows: List[List[str]] = []
+        if geostore_info is not None:
+            for info in geostore_info:
+                # Use the geostoreId itself as the id field for the output.
+                id = info["geostoreId"]
+                # The RW find-by-ids call returns the geometry as a feature collection,
+                # which I think should always have one feature (?)
+                fc = info["geostore"]["data"]["attributes"]["geojson"]["features"]
+                if fc is None:
+                    raise Exception(f"Missing features attribute for geostore '{id}'")
+                # GeoDataFrame.from_features() expects each feature to have a
+                # 'properties' field.
+                for f in fc:
+                    if f.get("properties") is None:
+                        f["properties"] = {}
+                minidf = gpd.GeoDataFrame.from_features(fc)
```

Comment:
if you're iterating one geostore at a time, I think you can skip turning it into a GDF and just call `shape(fc[0]["geometry"])`, which can covert GeoJSON to shapely directly